### PR TITLE
Multi line frame text

### PR DESCRIPTION
**Pull request by aditya-beaconstac**
This PR allows multi line frame text to be added to QR Codes

### CODE REVIEW

**Review by himanshu-km — APPROVED** [review]

File: src/Util.ts [hunk 1]
```diff
@@ -35,4 +35,11 @@ export const getFrameTextSize = (configSize: number, textLength: number) => {
         throw new Error('Frame text should be between 0 to 30 characters')
     }
     return configSize / factor;
-};
\ No newline at end of file
+};
+
```
Comment (aman-uniqode) [inline src/Util.ts:39]: this function is not readable to me and one more thing, why is the default value 7 any reason for that ?

File: src/Svg.ts [hunk 2]
```diff
@@ -177,23 +187,23 @@ export class SVGDrawing {
                     break;
                 case QRCodeFrame.BANNER_TOP:
                     this.shiftX = 1.5 * this.config.moduleSize;
-                    this.shiftY = 1.5 * this.config.moduleSize + size / 5 - 1;
+                    this.shiftY = (1.5 * this.config.moduleSize + size / 5 - 1) + multiLineHeight;
                     break;
                 case QRCodeFrame.BANNER_BOTTOM:
                     this.shiftX = 1.5 * this.config.moduleSize;
                     this.shiftY = 1.5 * this.config.moduleSize;
```
Comment (aman-uniqode) [inline src/Svg.ts:194]: why are we not adding multiLineHeight here, I am not aware of the the code but is this intended or we missed ?

File: src/Svg.ts [hunk 3]
```diff
@@ -177,23 +187,23 @@ export class SVGDrawing {
...
                     break;
                 case QRCodeFrame.BOX_TOP:
                     this.shiftX = 1.5 * this.config.moduleSize;
-                    this.shiftY = 2.5 * this.config.moduleSize + size / 5;
+                    this.shiftY = (2.5 * this.config.moduleSize + size / 5) + multiLineHeight;
                     break;
                 case QRCodeFrame.TEXT_ONLY:
```
Comment (aman-uniqode) [inline src/Svg.ts:200]: same for the TEXT_ONLY type,

File: src/Svg.ts [hunk 4]
```diff
@@ -2062,6 +2072,12 @@ export class SVGDrawing {
             // preloadFonts();
         }
 
+        const textLinesLength = text.length ? text.split('\n').length : 0;
+
+        const textLineMaxLength = getLengthOfLongestText(text);
+        const fontSize = getFrameTextSize(this.config.viewportSize, textLineMaxLength);
+
+        const multiLineHeight = ( textLinesLength - 1 ) * (fontSize + 10);
```
Comment (aman-uniqode) [inline src/Svg.ts:2080]: can I get an intuition of what we done here, why (fontSize+10) here?

**Review by aman-uniqode — COMMENTED** [review]
Left some comments, just some questions

**Pull request merged**